Add option "isassorted" to allow assorted values for select options

maurojs10 opened the pull request.

gjunge commented:
Hi Mauro,

Thanks for the pull request. 
Can you give me the use case for this new option?  
I want to see if it adds something for a decent amount of users, or if it has specific use for you. 

Thanks,
Gidon

maurojs10 commented:
Please have a look at this fiddle:

https://jsfiddle.net/maurojs10/3fspgg3f/

The backing field is a `select` and its `option` items have assorted values.

Those options are user-created but their values are auto-generated.

To be honest, I don't like this new option "isassorted". The code executed when "isassorted" is true also works with regular sorted values. So I think this new code could replace the original one without any configuration option.

Thank you very much for taking the time to consider this pull request.

gjunge commented:
I understand your use case, I also understand your comments. 
However we do have to keep our existing users in mind, and prevent from making breaking changes. 
Technically this would work today, and should work after the fix:
``` html
<select>
   <option value="0">None</option>  
   <option value="1">Bad</option>
   <option value="5">Barely</option>
   <option value="7">OK</option>
   <option value="10">Excellent</option>
</select>
```

I thought maybe to add a property to the backing field, since that is the "source of the troubles". 
Something like 
``` html
<select data-rateit-valuesrc="value|index">

</select>
```

What is your opinion?

maurojs10 commented:
I think your idea is great. Much better than my suggestion.

However, I'm afraid I don't get how to implement it.

 I'm not sure if this is what you have in mind, but I made this implementation: https://github.com/maurojs10/rateit.js/commit/8bd30179b61e1cad41da0ee1400604c447384676 . And also this new fiddle: https://jsfiddle.net/maurojs10/5pp6op1k/1/ .

And the [documentation](https://github.com/gjunge/rateit.js/wiki#options) says options can be set in different ways, but it says nothing about backingfield-only options. Would that be a new thing?

gjunge commented:
@maurojs10  that looks good. 
Yes it would be a new thing. Since it is such a specific thing for select fields only, it seems to be correct to add it as an option to the backing field itself. This way it becomes pretty self explanatory.

If you can add just some comments to the code would be great, and then just create a new pull request.
After that we'll add you Fiddle sample as a real example in the examples page, and add some documentation.

Thanks again!

maurojs10 commented:
Added comments: https://github.com/maurojs10/rateit.js/commit/b4e8a3f04a1abfe8f8225e172ca4dafa68fcf38e

Changed to plain JavaScript: https://github.com/maurojs10/rateit.js/commit/cfe407aefc70d43a9072d7d31ac6894d3591192d
(I thought it would be better to get backing field attribute via plain JavaScript instead of using jQuery.)

Created new pull request https://github.com/gjunge/rateit.js/pull/8

I'm closing this pull request.

Thank you very much.
